docs/developing.rst

```diff
@@ -0,0 +1,185 @@
+* Start out by trying a practice proctored exam to understand the process
+* The Instructor Dashboard has a "Special Exams" tab for administering proctoring
+    * can add allowances per user, e.g. additional time for an exam
+    * can also reset exam attempts for individual users
+* More debugging can be done in Django admin
+    * "Edx_Proctoring" section
+    * "Proctored exam software secure review" has log of responses from SoftwareSecure
+
+How do I develop on edx-proctoring?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+These are the steps to install edx-proctoring into your pre-existing devstack image:
+
+Clone edx-proctoring into the src directory next to edx-platform folder in your host filesystem::
+
+    $ cd src
+    $ git clone https://github.com/edx/edx-proctoring
+    Clone the mock proctoring server repo into your workspace:
+
+    $ cd src
+    $ git clone git@github.com:davestgermain/mockprock.git
+    Install the proctoring package and mockprock into edx-platform in the container, for both LMS and Studio:
+
+Edit or create the ``edx-platform/requirements/private.txt`` file::
+
+    # edx-platform/requirements/private.txt
+    # This file will be used in the docker image, so use file paths that work there.
+    -e /edx/src/edx-proctoring
+    -e /edx/src/mockprock
+
+Then::
+
+    $ make lms-shell
+    root@a7ff4f3f3f6b:/edx/app/edxapp/edx-platform# paver install_prereqs;exit
+    $ make studio-shell
+    root@a7ff4f3f3f6b:/edx/app/edxapp/edx-platform# paver install_prereqs;exit
+
+
+You can also develop edx-proctoring outside of devstack with a virtual environment::
+
+    cd edx-proctoring
+    mkvirtualenv edx-proctoring
+    make upgrade requirements
+    make test-python
+
+In edx-platform/lms/envs/private.py and edx-platform/cms/envs/private.py:
+
+.. code-block:: python
+
+    from .production import FEATURES
+     
+    FEATURES['ENABLE_SPECIAL_EXAMS'] = True
+     
+    PROCTORING_BACKENDS = {
+        'DEFAULT': 'mockprock',
+        'null': {},
+        'mockprock': {
+            'client_id': 'abcd',
+            'client_secret': 'abcdsecret',
+            'default_rules': {
+                'allow_snarfing': True,
+                'allow_grok': False
+            }
+        }
+    }
+ 
+    PROCTORING_SETTINGS = {
+        'MUST_BE_VERIFIED_TRACK': False
+    }
+
+Then back in your host shell::
+
+    cd ~/workspace/mockprock/
+    pip install -e .[server]
+    python -m mockprock.server
+
+The command will tell you you have to supply an client_id and client_secret. It'll open your browser to the django admin page where you should create or use an existing credential. Note the client_id and client_secret and restart the server::
+
+    python -m mockprock.server {client_id} {client_secret}
+
+How do I debug proctoring tests?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Set up a virtual environment, then::
+
+    cd /edx/src/edx-proctoring
+    make test-all
```

Comment:
This `make` command uses `tox`, so there shouldn't be any benefit to setting up a venv as this suggests a few lines above

Reply:
(I think we might need to replace most of this section; some of this stuff looks like it was outdated before we started this project.)

Reply:
I replaced the "debugging tests" section with this brief "running tests" section